src/modules/loadPage.js

```diff
@@ -1,98 +1,68 @@
-import { classify, createHistoryRecord, fetch } from '../helpers.js';
+import { classify, createHistoryRecord, updateHistoryRecord, fetch, getCurrentUrl } from '../helpers.js';
 
-const loadPage = function(data, popstate) {
-	// create array for storing animation promises
-	let animationPromises = [],
-		xhrPromise;
-	const animateOut = () => {
-		this.triggerEvent('animationOutStart');
+const loadPage = function(data, popstate = false) {
```

Comment:
Would it make sense to rewrite this to an async function, so we could handle calling and awaiting both `leavePage` and `renderPage` inside of here? Maybe it could put all the handling of classes nicely into one place. 
Right now (historically) it seems like it could be unclear why are some classes added/removed at different times.

Reply:
I'd love to rewrite everything with `async/await`, but that's one step too high on the complexity ladder for my taste. I'd prefer to get the functionality working and tackle async-await for the whole codebase as a separate refactor PR.